modules/ServerAPI/src/server/routes/LabelServiceRoutes.coffee

```diff
@@ -9,21 +9,25 @@ exports.getNextLabelSequence = (req, resp) ->
 		labelServiceTestJSON = require '../public/javascripts/spec/testFixtures/LabelServiceTestJSON.js'
 		resp.json labelServiceTestJSON.nextLabelSequenceResponse
 	else
-		config = require '../conf/compiled/conf.js'
-		baseurl = config.all.client.service.persistence.fullpath+"labelsequences/getLabels"
-		request = require 'request'
-		request(
-			method: 'POST'
-			url: baseurl
-			body: req.body
-			json: true
-		, (error, response, json) =>
-			if !error && response.statusCode == 200
-				resp.json json
-			else
-				console.log 'got ajax error trying to get next label sequence'
-				console.log error
-				console.log json
-				console.log response
-				resp.end JSON.stringify "getNextLabelSequenceFailed"
-		)
+		exports.getNextLabelSequenceInternal req.body, (callback) ->
+			resp.json callback
+
+exports.getNextLabelSequenceInternal = (labelSequence, callback) ->
+	config = require '../conf/compiled/conf.js'
+	baseurl = config.all.client.service.persistence.fullpath+"labelsequences/getLabels"
+	request = require 'request'
+	request(
+		method: 'POST'
+		url: baseurl
+		body: labelSequence
+		json: true
+	, (error, response, json) =>
+		if !error && response.statusCode == 200
+			callback json
+		else
+			console.log 'got ajax error trying to get next label sequence'
+			console.log error
+			console.log json
+			console.log response
+			callback "getNextLabelSequenceFailed"
```

Comment:
this one, gets a call back no matter what. which is good. See other places where it doesn't